### PR TITLE
[t3][p0L][1/n] request API

### PR DESCRIPTION
**Pull request by rl3250**
### Goal
This is the sequence of implementation of request API.
### Solution
1. implement `transactionService`, which is the only service that manipulates the data
2. implement `requestService` which handles request related apis, and it calls `transactionService` to execute stuff
3. controller - create new request

### In this PR
1. I delivered 1, 2, 3 in this PR. 
2. Some refactoring needed but they are not blockers.
### Testing
(3) is being tested using postman and requests are written to local DB:
image
image

### In next PR
1. deliver controller for retrieve all user's requests
4. deliver controller for accepting requests

CC: @howieraem @jenniferduan45 @KenXiong123

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,8 +1,8 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/4156db
 spring.datasource.username=root
-spring.datasource.password=
+spring.datasource.password=dbuserdbuser
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.database-platform = org.hibernate.dialect.MySQL8Dialect
 spring.jpa.generate-ddl=true
-spring.jpa.hibernate.ddl-auto = create
+spring.jpa.hibernate.ddl-auto = update
```
Comment (howieraem) [inline src/main/resources/application.properties:7]: Let's keep spring.jpa.hibernate.ddl-auto = create for now, as we may alter the schema quite often recently

File: src/main/java/com/lgtm/easymoney/payload/RequestRsp.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import java.math.BigDecimal;
+
+@AllArgsConstructor
+@Data
+@NoArgsConstructor
+public class RequestRsp {
```
Comment (jenniferduan45) [inline src/main/java/com/lgtm/easymoney/payload/RequestRsp.java:11]: Though this is the same as BalanceRsp, I wonder if it is better to still keep RequestRsp in order to maintain consistency in payload format and naming convention.

File: src/main/java/com/lgtm/easymoney/payload/RequestReq.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private Long to_uid;
+
+    @NotNull
+    @Digits(integer = 100, fraction = 2)
+    @DecimalMin(value = "0.0", inclusive = false)
+    private BigDecimal amount;
+
+    private String description;
+
+    private Category category;
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/payload/RequestReq.java:25]: Did you test with string in Postman?

File: src/main/java/com/lgtm/easymoney/payload/RequestReq.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import javax.validation.constraints.Digits;
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+@Data
+public class RequestReq {
+    @NotNull
+    private Long from_uid;
+    @NotNull
+    private Long to_uid;
```
Comment (howieraem) [inline src/main/java/com/lgtm/easymoney/payload/RequestReq.java:16]: Keep variable and function naming in lower camel case without underscores?

**Review by howieraem — COMMENTED** [review]
Mostly lg, left some comments

File: src/main/java/com/lgtm/easymoney/payload/RequestReq.java (hunk 3 above)
Comment (rl3250) [reply src/main/java/com/lgtm/easymoney/payload/RequestReq.java:25]: yes

File: src/main/java/com/lgtm/easymoney/services/impl/RequestServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    @Override
+    public ResponseEntity<BalanceRsp> createARequest(RequestReq req) {
+        // get params
+        Long from_uid = req.getFrom_uid();      // from = req by
+        Long to_uid = req.getTo_uid();          // to = req to
+        BigDecimal amount = req.getAmount();
+        Category category = req.getCategory();
+        String desc = req.getDescription();
+        // make a deposit
```
Comment (jenniferduan45) [inline src/main/java/com/lgtm/easymoney/services/impl/RequestServiceImpl.java:75]: Is this make a deposit? Probably make a request?

Comment (rl3250) [reply src/main/java/com/lgtm/easymoney/services/impl/RequestServiceImpl.java:75]: yeah you are right, copied the comment, will fix this nits

**Comment by rl3250** [comment]
addressed comments from @howieraem  @jenniferduan45

**Review by KenXiong123 — APPROVED** [review]

**Review by howieraem — APPROVED** [review]

**Review by jenniferduan45 — APPROVED** [review]
lgtm now

**Pull request merged**